### PR TITLE
Update of documentation for DAX_KMEM kinds

### PR DESCRIPTION
**Pull request by kasiawasiuta**
Update of memkind.3 man pages to describe the differences between DAX_KMEM kinds.
### Types of changes

- [ ] Bugfix (non-breaking change which fixes issue linked in Description above)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation (correction or otherwise)
- [ ] Cosmetics (whitespace, appearance)
- [ ] Other

### Checklist

- [ ] Code compiles without errors
- [ ] All tests pass locally with my changes (see TESTING section in CONTRIBUTING file)
- [ ] Created tests which will fail without the change (if possible)
- [x] Extended the README/documentation (if necessary)
- [ ] All newly added files have proprietary license (if necessary)
- [ ] All newly added files are referenced in MANIFEST files (if necessary)

## Further comments

---
This change is [Reviewable](https://reviewable.io/reviews/memkind/memkind/309)

### CODE REVIEW

File: man/memkind.3 [hunk 1]
```diff
@@ -748,30 +748,34 @@ across all high bandwidth nodes and transparent huge pages are
 disabled.
 .TP
 .B MEMKIND_DAX_KMEM
-Allocate from the closest persistent memory NUMA node(s) at time
-of allocation. If there is not enough memory in closest persistent memory NUMA node(s) to satisfy the request
+Allocate from the closest persistent memory NUMA node at time
+of allocation. If there is not enough memory in closest avaliable persistent memory NUMA node to satisfy the request
```
Comment (marcinslusarz) [inline man/memkind.3]: available

File: man/memkind.3 [hunk 2]
```diff
@@ -748,30 +748,34 @@ across all high bandwidth nodes and transparent huge pages are
...
 and the allocated pointer is set to
 .IR "NULL" .
 .TP
 .B MEMKIND_DAX_KMEM_ALL
-Allocate from the closest persistent memory NUMA node available at the time
-of allocation. If there is not enough memory on any of persistent memory NUMA nodes to satisfy the request
+Same as
+.B MEMKIND_DAX_KMEM
+except that if there is not enough memory in the closest persistent memory NUMA node to satisfy
+ the request, tha allocation will fall back on next closest persistent memory numa node(s), if avaliable.
```
Comment (marcinslusarz) [inline man/memkind.3]: fall back **to the** next...
available

Comment (kasiawasiuta) [reply man/memkind.3]: I fixed "the available mistake", but please take a look at the explanation of the "fall back on" phrase
in the dictionary: [macmillandictionary.com](https://www.macmillandictionary.com/dictionary/british/fall-back-on).

File: man/memkind.3 (hunk 1 above)
Comment (kasiawasiuta) [reply man/memkind.3]: Done :)

**Pull request merged**